x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/ingest/SetSecurityUserProcessor.java

```diff
@@ -93,7 +96,11 @@ public IngestDocument execute(IngestDocument ingestDocument) throws Exception {
             switch (property) {
                 case USERNAME:
                     if (user.principal() != null) {
-                        userObject.put("username", user.principal());
+                        if (escCompliant && "user".equals(field)) {
```

Comment:
This is not ideal.

It means if you set 
`{ "field": "user", "ecs_compliant": true}` you get a `name` field but for `{ "field": "es_user", "ecs_compliant": true}` you get a `username` field. That's unnecessarily confusing .

My gut-feel preference would be that `ecsCompliant` mode _requires_ that the field be named `"user"`.
`{ "field": "auth", "ecs_compliant": true}` should fail as an invalid configuration.

Reply:
You are right. It is very confusion. The option is effectively just a "hint". I think tighten the behaviour as you suggested makes sense. Will update. Thanks